### PR TITLE
add kerberos for druid

### PR DESCRIPTION
**Pull request by pokyu**
add kerberos configure for Druid

### CODE REVIEW

**Comment by pokyu** [comment]
what can`t be done ?

**Comment by pokyu** [comment]
？

File: superset/connectors/druid/models.py [hunk 1]
```diff
@@ -262,6 +272,17 @@ def name(self):
...
 
+    def get_kerberos_auth(self):
+        enable_kerberos_authentication = conf.get('ENABLE_KERBEROS_AUTHENTICATION', False)
+        kerberos_keytab = conf.get('KERBEROS_KEYTAB', [])
+        kerberos_principal = conf.get('KERBEROS_PRINCIPAL', [])
+        if not enable_kerberos_authentication:
+            return False
+
+        kerberos_commands = 'kinit -k -t %s %s' % (kerberos_keytab, kerberos_principal)
+        subprocess.call(kerberos_commands, shell=True)
```
Comment (craig-rueda) [inline superset/connectors/druid/models.py]: Not sure I feel comfortable shelling out like this, from within a model, no less. This sort of logic belongs in a higher layer (connection manager, perhaps?)

File: superset/connectors/druid/models.py [hunk 2]
```diff
@@ -147,11 +150,18 @@ def get_pydruid_client(self):
...
+        enable_kerberos_falg = conf.get('ENABLE_KERBEROS_AUTHENTICATION', False)
+        if enable_kerberos_falg:
+            kerberos_auth = HTTPKerberosAuth(mutual_authentication=OPTIONAL)
+            response = requests.get(endpoint, auth=kerberos_auth)
+        else:
+            response = requests.get(endpoint)
+        return json.loads(response.text)
 
     def get_druid_version(self):
+
```
Comment (craig-rueda) [inline superset/connectors/druid/models.py]: extra space?

File: requirements.txt [hunk 3]
```diff
@@ -72,3 +72,4 @@ webencodings==0.5.1       # via bleach
 werkzeug==0.14.1          # via flask
 wtforms-json==0.3.3
 wtforms==2.2.1            # via flask-wtf, wtforms-json
+requests-kerberos==0.12.0
```
Comment (craig-rueda) [inline requirements.txt]: This should be in extras

**Review by craig-rueda — CHANGES_REQUESTED** [review]
Needs a few tests around Kerberos stuff. Ideally, the fetching of datasources would be broken out of the "model" and would live in another layer.

**Comment by elukey** [comment]
Any news on this pull request?

**Comment by craig-rueda** [comment]
Looks like there's merge conflicts, and there were a few comments that weren't addressed.

File: requirements.txt (hunk 3 above)
Comment (mistercrunch) [reply requirements.txt:75]: +1 let's create a kerberos group in extra_requires

**Comment by mistercrunch** [comment]
FYI, the Druid connector is getting deprecated [slowly over time] in favor of using SQLAlchemy and the SQL interface to Druid. We can still merge this once the comments are addressed, but the core committers will fade out of supporting the native Druid connector. This aligns with the Druid roadmap where they're clearly doubling down on the SQL interface

**Pull request closed without merging**